dataswarm/manager/dataswarm_file.c

```diff
@@ -0,0 +1,51 @@
+
+#include "dataswarm_file.h"
+#include "helpers.h"
+
+#include <string.h>
+
+int dataswarm_declare_file(struct jx *json){
+    
+    char *type = NULL;
+    int project = 0;
+    char *metadata = NULL;
+    
+    //validate json
+    if(validate_json(json, declaration)){
+        return 0;
+    }
+
+    //get file info from jx struct --> type, project, metadata
+    void *i = NULL;
+    void *j = NULL;
+    const char *key = jx_iterate_keys(json, &j);
+    struct jx *value = jx_iterate_values(json, &i);
+
+    while(key != NULL){
+        
+        if(!strcmp(key, "type")){
+            type = value->u.string_value;
+        } else if(!strcmp(key, "project")){
+            project = value->u.integer_value;
+        } else if(!strcmp(key, "metadata")){
+            metadata = value->u.string_value;
+        }
+
+    }
+
+    // assign a UUID to the file
+    // TODO: create vv function
+    int uuid = dataswarm_assign_uuid();
```

Comment:
For future generality, a uuid needs to be a string, since it may have a form like this:
https://en.wikipedia.org/wiki/Universally_unique_identifier

Also see `dttools/src/uuid.h` for our local implementation.

Reply:
should it be of type string or of type cctools_uuid_t?

Reply:
Generate a UUID, but pass it around as a plain string with no special interpretation.  The worker just knows that the manage r refers to blobs by name, and the naming scheme is up to the manager.  likewise, the application shouldn't make any limiting assumptions about the file name.